nextjs-app/components/common/SloganPopup/SloganPopup.tsx

```diff
@@ -0,0 +1,67 @@
+"use client";
+
+import { FC, useEffect, useState, type ReactNode } from "react";
+import { twMerge } from "tailwind-merge";
+import { SloganProvider, useSlogan } from "./slogan-context";
+import { default as PopupPointerImg } from "./assets/popup-pointer.svg";
+
+interface SloganProps {
+  children: ReactNode;
+  slogans: string[];
+}
+
+const SloganContainer: FC<SloganProps> = ({ children, slogans }) => {
+  return <SloganProvider slogans={slogans}>{children}</SloganProvider>;
+};
+
+interface SloganItemProps {
+  slogan: string;
+  position?: "left" | "right";
+  className?: string;
+}
+
+const SloganPopup: FC<SloganItemProps> = ({
+  slogan,
+  position = "right",
+  className,
+}) => {
+  const [isActive, setIsActive] = useState(false);
+  const { randomSlogans } = useSlogan();
+
+  useEffect(() => {
+    if (Array.isArray(randomSlogans) && randomSlogans.includes(slogan)) {
```

Comment:
NIT: if we can return a Set from the provider, maybe the logic to search the slogan can be reduce from O(n) to O(1)
But the data is small here, so I don't think it will make big different